checkers/src/Pawn.cpp

```diff
@@ -1,8 +1,61 @@
 #include "Pawn.hpp"
 #include "Board.hpp"
+#include "Move.hpp"
+#include <vector>
 
-bool Pawn::canCapture(Piece& piece, Board &board) {
-    if (this->isWhite() == piece.isWhite()) return false;
+void Pawn::getCaptureMoves(std::vector<Move> &moves, Board board, Move current) {
+
+    int dx[4] = {1,1,-1,-1};
+    int dy[4] = {-1,1,-1,1};
+
+    int countMoves = 0;
+    for(int i = 0; i <4; ++i){
+        Position pos(pos_.x + dx[i], pos_.y + dy[i]);
+        if (pos.x >=0 && pos.y>= 0) {
```

Comment:
To jest warunek na poprawne zawieranie się w tablicy? bo jeszcze zostaje wtedy górne porównanie <=7
i jest metoda Position::isPositionValid(), która to może zrobić (w sumie można by jej nazwę zmienić na isValid, bo wiadomo, że to Position)